### ISSUE TITLE
Infinite loop in the affiliate links example

### ISSUE THREAD

**Issue by dvoytenko** [WICG/app-history#43]
> Example: cross-origin affiliate links

```
appHistory.addEventListener("navigate", e => {
  const url = new URL(e.destination.url);
  if (url.hostname === "store.example.com") {
    url.queryParams.set("affiliateId", "ead21623-781e-442f-a2c4-6cc1b2a9fda2");

    e.preventDefault();
    location.href = url;
  }
});
```

Does this introduce an infinte loop? Should it also check that the `affiliatedId` is already set?

**Comment by domenic** [comment]
Yeah, that's quite busted. We want to fix that as part of #5.

**Comment by domenic** [comment]
Let me merge this into #5.
